Flaky test: org.keycloak.operator.testsuite.integration.ClusteringTest#testKeycloakScaleAsExpected

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

operator

### Describe the bug

It seems that this test is unstable. In addition, the test doesn't log much useful information. See below for a link with the full details.

```
2023-06-09T13:00:48.0157656Z ------- STARTING: ClusteringTest.testKeycloakScaleAsExpected()
2023-06-09T13:00:48.0158336Z ------- Namespace: keycloak-test-31615d53-c556-4f7c-a8e2-0fe4185a68a2
2023-06-09T13:00:48.0158901Z ------- Mode: remote
2023-06-09T13:00:48.1156337Z Jun 09, 2023 1:00:48 PM org.keycloak.operator.testsuite.utils.K8sUtils waitForKeycloakToBeReady
2023-06-09T13:00:48.1157119Z INFO: Waiting for Keycloak "example-kc"
2023-06-09T13:03:17.2859565Z Jun 09, 2023 1:03:17 PM org.keycloak.operator.testsuite.integration.BaseOperatorTest cleanup
2023-06-09T13:03:17.2860327Z INFO: Deleting Keycloak CR
2023-06-09T13:03:18.3872058Z Jun 09, 2023 1:03:18 PM org.keycloak.operator.testsuite.integration.BaseOperatorTest cleanRBACresourcesAndOperatorDeployment
2023-06-09T13:03:18.3873096Z INFO: Deleting RBAC from Namespace keycloak-test-31615d53-c556-4f7c-a8e2-0fe4185a68a2
2023-06-09T13:03:18.4872188Z Jun 09, 2023 1:03:18 PM org.keycloak.operator.testsuite.integration.BaseOperatorTest after
2023-06-09T13:03:18.4873130Z INFO: Deleting namespace : keycloak-test-31615d53-c556-4f7c-a8e2-0fe4185a68a2
2023-06-09T13:03:18.5990658Z [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 402.697 s <<< FAILURE! - in org.keycloak.operator.testsuite.integration.ClusteringTest
2023-06-09T13:03:18.5991983Z [ERROR] org.keycloak.operator.testsuite.integration.ClusteringTest.testKeycloakScaleAsExpected  Time elapsed: 150.36 s  <<< ERROR!
2023-06-09T13:03:18.5992896Z org.awaitility.core.ConditionTimeoutException: 
2023-06-09T13:03:18.5993547Z Assertion condition defined as a org.keycloak.operator.testsuite.integration.ClusteringTest 
2023-06-09T13:03:18.5994066Z expected: 3
2023-06-09T13:03:18.5994519Z  but was: 2 within 1 minutes.
2023-06-09T13:03:18.5995016Z    at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:164)
2023-06-09T13:03:18.5995618Z    at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
2023-06-09T13:03:18.5996283Z    at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
2023-06-09T13:03:18.5997160Z    at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:939)
2023-06-09T13:03:18.5997682Z    at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:723)
2023-06-09T13:03:18.5998312Z    at org.keycloak.operator.testsuite.integration.ClusteringTest.testKeycloakScaleAsExpected(ClusteringTest.java:98)
2023-06-09T13:03:18.5998923Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-06-09T13:03:18.5999439Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-06-09T13:03:18.6000008Z    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-06-09T13:03:18.6000478Z    at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2023-06-09T13:03:18.6001176Z    at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:1014)
2023-06-09T13:03:18.6001792Z    at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:828)
2023-06-09T13:03:18.6002465Z    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
2023-06-09T13:03:18.6003253Z    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
2023-06-09T13:03:18.6004161Z    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
2023-06-09T13:03:18.6004819Z    at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
2023-06-09T13:03:18.6005420Z    at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
2023-06-09T13:03:18.6006037Z    at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
2023-06-09T13:03:18.6006717Z    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
2023-06-09T13:03:18.6007433Z    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
2023-06-09T13:03:18.6008162Z    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
2023-06-09T13:03:18.6008837Z    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
2023-06-09T13:03:18.6009495Z    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
```

### Version

nightly

### Expected behavior

The tests should fail. If it fails, it should contain more actionable information (log, events, Kubernetes resources)

### Actual behavior

Test fails randomly, and don't reveal information in the log which could have helped me to find out what went wrong.

### How to Reproduce?

Unknown - see https://github.com/keycloak/keycloak/actions/runs/5221959664/jobs/9427086805?pr=20883 for a full run

### Anything else?

_No response_

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/ClusteringTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/ClusteringTest.java
@@ -92,11 +92,6 @@ public class ClusteringTest extends BaseOperatorTest {
                 .ignoreExceptions()
                 .untilAsserted(() -> assertThat(kcPodsSelector.list().getItems().size()).isEqualTo(3));
 
-        Awaitility.await()
-                .atMost(Duration.ofSeconds(60))
-                .ignoreExceptions()
-                .untilAsserted(() -> assertThat(crSelector.scale().getStatus().getReplicas()).isEqualTo(3));
-
         // when scale it down to 2
         crSelector.scale(2);
         assertThat(crSelector.scale().getSpec().getReplicas()).isEqualTo(2);

